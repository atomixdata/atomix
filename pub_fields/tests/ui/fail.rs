// put the struct in its own module
mod inner {
    #[cfg_attr(test, pub_fields::pub_fields)]
    pub struct Bar {
        a: i32, // private field if cfg(test)=false
    }
}

// crate-root (a sibling module) tries to construct it
fn main() {
    let _b = inner::Bar { a: 0 };
    // ^^^^^  error[E0451]: field `a` of struct `inner::Bar` is private
}
