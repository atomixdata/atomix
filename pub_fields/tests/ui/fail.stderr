error[E0451]: field `a` of struct `Bar` is private
  --> tests/ui/fail.rs:11:27
   |
11 |     let _b = inner::Bar { a: 0 };
   |                           ^^^^ private field
